src/test/java/stashpullrequestbuilder/stashpullrequestbuilder/StashRepositoryTest.java

```diff
@@ -437,4 +437,38 @@ public void startJobSkipsNullParameters() {
 
     assertThat(parameters, is(empty()));
   }
+
+  @Test
+  public void parameters_populated_from_environment() {
+    cause = makeCause(null);
+
+    ParameterDefinition parameterDefinition =
+        new StringParameterDefinition("sourceBranch", "DefaultBranch");
+    jobSetup(parameterDefinition);
+
+    List<ParameterValue> parameters = captureBuildParameters();
+
+    assertThat(parameters, hasSize(1));
+    assertThat(parameters.get(0), instanceOf(StringParameterValue.class));
+    assertThat(parameters.get(0).getName(), is("sourceBranch"));
+    assertThat(parameters.get(0).getValue(), is("SourceBranch"));
```

Comment:
It's a bit unclear where the value `SourceBranch` comes from. I think it's from line 95.
Maybe extracting that into a constant would make it easier for the readers?

Reply:
You are right. I'm not sure how to do it better. Having variables for all arguments to the `StashCause` constructor would add a lot of code of little utility. Using a variable only for `sourceBranch` would be inconsistent and would suggest that we are treating it in a special way.

I can use names suggesting the branch, such as `source/branch` and `target/branch`, but again, I don't have a good idea how to do it consistently for arguments like `SourceRepositoryOwner`.

Leaving it as is for now.

Reply:
Usually I would try to use a builder, changing the line `  cause = makeCause(prParameters);`
to something like this:
```
 cause = defaultCauseBuilder()
                .withAdditionalParameters(prParameters)
                .withSourceBranch("source branch from cause")
                .build();
```

Reply:
I'm not insisting on change it now though, the code is still better than it was before

Reply:
The builder pattern is not a good match for `StashCause`, as it has no reasonable default values for anything other than the additional parameters. The builder would not be useful in the production code.

I replaced `"SourceBranch"` with `cause.getSourceBranch()`, that should make the intention clear.